tembo-operator/src/network_policies.rs

```diff
@@ -39,42 +39,68 @@ pub async fn reconcile_network_policies(client: Client, namespace: &str) -> Resu
         "apiVersion": "networking.k8s.io/v1",
         "kind": "NetworkPolicy",
         "metadata": {
-          "name": "allow-egress-to-kube-dns",
-          "namespace": format!("{namespace}"),
+            "name": "allow-egress-to-gke-dns",
```

Comment:
Is there a reason for the name change? It looks like we're adding one egress entry for GKE but the existing entry for kube-dns is still there.

Reply:
The GKE clusters are built with node-local-dns enabled.  This runs as a `DaemonSet` on each node and intercepts DNS traffic to `kube-dns`.  So the policy was modified to allow traffic to either `node-local-dns` or allow traffic to `kube-dns` both running in the `kube-system` namespace.

This allows the policy to work for both EKS and GKE (with or without node-local-dns enabled).

Reply:
That makes sense. I think it we should give this netpol a more generic name as it will be present on GKE, EKS and eventually AKS clusters. What do you think?

Reply:
I will add, if we do need to disable `node-local-dns` for any reason then the policy will still work as expected, and we can just remove it later.  There are pro's and con's for running it and not running it.  `kube-dns` has some limitations (20 open connections per pod).  GKE does auto scale `kube-dns` but that can lead to other issues at scale.  `node-local-dns` takes that load off of `kube-dns` under heavy load and will cache DNS queries.  This leads to faster DNS responses, but sometimes can cause other issues on larger clusters.  

After some reading if this happens, we should migrate to using CloudDNS with GKE which I haven't seen before, but it's the recommended path https://cloud.google.com/kubernetes-engine/docs/how-to/cloud-dns.  This comes with some added cost to running the GKE cluster as well.

Reply:
Oh I see what you're talking about @ianstanton.  I will make that change to make it more generic.